lib/cloudregister/registerutils.py

```diff
@@ -605,10 +612,90 @@ def set_registry_auth_token(registry_fqdn, username, password):
         list(config_json.get('auths', {}).items()) +  # old content
         list(registry_credentials.items())            # new content
     )
-    updated = write_registry_credentials(config_json)
+    updated = write_registry_credentials(content=config_json, set_new=True)
     return updated
 
 
+# ----------------------------------------------------------------------------
+def clean_registry_setup():
+    """Remove the data previously set to make the registry work."""
+    clean_registry_auth()
+
+
+# ----------------------------------------------------------------------------
+def clean_registry_auth():
+    """Clean the auth token from the config json file."""
+    config_json, unset_failed = get_registry_credentials(set_new=False)
+    if (
+        (not config_json or config_json == {'auths': {}}) and
+        (unset_failed is None or unset_failed == 0)
+    ):
+        # credentials file is empty or
+        # credentials file is not empty but its content is useless or
+        # could not access the file but the backup suceeded
+        if unset_failed is None:
+            logging.info('JSON content is empty')
+            os.unlink(REGISTRY_CREDENTIALS_PATH)
+        return True
+
+    if unset_failed:
+        return False
+
+    # we could open the credentials file
+    # and it is not empty
+    # unset the registry credentials
+    entry = None
+    smt = get_smt_from_store(__get_registered_smt_file_path())
+    if same_registry_auth_content(config_json, smt):
+        # if the content of the registry auth file is only
+        # our auth info, remove the file
+        logging.info(
+            'The content of the registry auth is our credentials only. '
+            'Removing the file %s' % REGISTRY_CREDENTIALS_PATH
+        )
+        os.unlink(REGISTRY_CREDENTIALS_PATH)
+        return True
+
+    if smt and smt.get_registry_FQDN():
+        logging.info(
+            'Unsetting the auth entry for %s' % smt.get_registry_FQDN()
+        )
+        entry = smt.get_registry_FQDN()
+    else:
+        logging.info('Unsetting the auth entry based on the token')
+        auth_token = __generate_registry_auth_token()
+        entries = config_json.get('auths', {}).items()
+        entry = [entry for entry, auth in entries if auth == auth_token]
```

Comment:
I think in [this case,](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/162/files#diff-84460adc47c95f16ae220daac1882a7bc19b595d1ba01650e286b666ef14f2a4R696) checking the content on a dict like `{"auths": {"xxxxx?": "auth_token_we_can_match"}}`

I think for this case, it would be better to check the parts we know of the FQDN, will modify it for that

Reply:
Done